lib/puppet/provider/razor_hook/rest.rb

```diff
@@ -0,0 +1,72 @@
+require File.join(File.dirname(__FILE__), '..', 'razor_rest')
+
+Puppet::Type.type(:razor_hook).provide :rest, :parent => Puppet::Provider::Rest do
+  desc "REST provider for Razor hook"
+  
+  mk_resource_methods
+  
+  def flush    
+    if @property_flush[:ensure] == :absent      
+      delete_hook
+      return 
+    end
+    
+    if @property_flush[:ensure] == :present      
+      create_hook
+      return 
+    end
+    
+    update_hook
+  end  
+
+  def self.instances
+    get_objects(:hooks).collect do |object|
+      new(object)
+    end
+  end
+  
+  # TYPE SPECIFIC
+  def self.get_object(name, url)
+    responseJson = get_json_from_url(url)    
+
+    {
+      :name           => responseJson['name'],
+      :hook_type      => responseJson['hook_type'],
+      :configuration  => responseJson['configuration'],
+      :ensure         => :present
+    }
+  end
+  
+  def self.get_hook(name)
+    rest = get_rest_info
+    url = "http://#{rest[:ip]}:#{rest[:port]}/api/collections/hooks/#{name}" 
```

Comment:
This URL should be derived from the entrypoint (via GET on `http://#{rest[:ip]}:#{rest[:port]}/api`, parsing that JSON, then looking up the `hooks` collection endpoint). That's part of the API contract and allows us to move endpoints if ever necessary.